### ISSUE TITLE
Issue when updating to HA 2022.7.0

### ISSUE THREAD

**Issue by Dillton** [fsaris/home-assistant-awox#59]
Hello @fsaris.

I have discovered a new issue when updating HA to 2022.7.0.
Unfortunatelly I have no logs because I have immediatelly rolled back to 2022.6.7 and aftetr rollback everything is fine.
Best guess, its something related to new Python 3.10 implemented in 2022.7.0.

Did you updated your HA to 2022.7.0? Can you please check what is broken after update?
Anyone else experiencing the same issue?

BR,
Michal

**Comment by alexvdleek** [comment]
Same issue here after update to 2022.7
No logbook entry found.
Says: Became unavailable.
2 Awox smart plugs.

**Comment by pipacsba** [comment]
Hi,

as I see the issue is that bluepy is not competible with 3.10.
for more see [https://github.com/IanHarvey/bluepy/issues/478]

Is there any suitable alternative?

Best,
Agoston

**Comment by OkOtChA666** [comment]
Same issue

**Comment by fsaris** [comment]
Sorry no solution yet.

I see that the core also uses bluepy for some components not sure how they handle that currently
https://github.com/home-assistant/core/search?q=bluepy&type=

**Comment by Dillton** [comment]
Good evening @fsaris.

Did you have a time to check new approach or do you have already any plan for migration to new libraries?
Can we help you somehow?

BR,
Michal

**Comment by fsaris** [comment]
Need to check if https://github.com/fsaris/home-assistant-awox/blob/main/custom_components/awox/awoxmeshlight/__init__.py can be updated to use bleak.

I definitely want to get my light's working with the newer HA version. But will not be able to look into this for the next 5 weeks. Have a road trip planned and currently busy with the preparations.

**Comment by dabepp** [comment]
> Have a road trip planned and currently busy with the preparations.

Enjoy the trip and come back in one piece! ;-)

**Comment by fsaris** [comment]
_Note for self_

Some examples of an bleak implementation as reference:
- https://github.com/emlove/pyzerproc/blob/main/pyzerproc/light.py
- https://github.com/emlove/pykulersky/blob/main/pykulersky/light.py
- https://github.com/hbldh/bleak/blob/develop/examples/async_callback_with_queue.py

**Comment by SergioGomera** [comment]
> _Note for self_
> 
> Some examples of an bleak implementation as reference:
> 
> * https://github.com/emlove/pyzerproc/blob/main/pyzerproc/light.py
> * https://github.com/emlove/pykulersky/blob/main/pykulersky/light.py
> * https://github.com/hbldh/bleak/blob/develop/examples/async_callback_with_queue.py

Hello fsaris, I hope you enjoy the roadtrip(where do you go??)  have you been able to make any progress with the integration?? Thx in advance.

**Comment by fsaris** [comment]
First steps are taking this evening. I'm able to use `bleak` to connect with the mesh and fetch the current status.

Hope to make some more progress during the weekend.

**Comment by LaTrappe** [comment]
Just a comment to follow this thread and to give @fsaris kudos for this.

**Comment by Nicolas1847** [comment]
Same, thank you for your work @fsaris

**Comment by fsaris** [comment]
Slowly making progress. Was able to turn on/off a light using the bleak library and ready the current status. 

But not able yet to trigger startNotify and control the light simultaneously. Looks like the light doesn't give the expected result that bluez/bleak expects.

**Comment by 0546timm** [comment]
Hey fsaris,

Many thanks for your work. Waiting eagerly on the update for my 'keukentafel' (kitchentable) lights!

**Comment by fsaris** [comment]
I created a beta release could you maybe test this (see  https://hacs.xyz/docs/faq/beta)?

I couldn't change to bleak (yet) as I'm running into an issue that the lights do not seem to follow the BLE standard (https://github.com/hbldh/bleak/issues/972).

That's why I switched to pygatt for now.

**Comment by OkOtChA666** [comment]
> I created a beta release could you maybe test this (see https://hacs.xyz/docs/faq/beta)?
> 
> I couldn't change to bleak (yet) as I'm running into an issue that the lights do not seem to follow the BLE standard ([hbldh/bleak#972](https://github.com/hbldh/bleak/issues/972)).
> 
> That's why I switched to pygatt for now.

Going to test this tomorrow. Thanks

**Comment by 0546timm** [comment]
Hey Frans,

I've downloaded it and tested, but unforatuntly the state of the light entities stays off and is not switchable to on.
It can find all the lamps and the addon installs fine, but they are not controlable.

https://i.gyazo.com/2dfc7eacb88a3faeb99f7f86934683bb.png

https://i.gyazo.com/d87cc96c109f40f5f0b117ecdd04b8f0.png

**Comment by alexvillehart** [comment]
Tested with a single light (ESpot 170) works fine for me, including states.
Logs do show a error, but as far as I've seen it doesn't affect usage, i've had this error before running hassio containerized on a unraid box.
`Find devices process error: Command 'PATH=/usr/sbin:$PATH; rfkill unblock bluetooth' returned non-zero exit status 1."`

**Comment by fsaris** [comment]
> > I created a beta release could you maybe test this (see https://hacs.xyz/docs/faq/beta)?
> > I couldn't change to bleak (yet) as I'm running into an issue that the lights do not seem to follow the BLE standard ([hbldh/bleak#972](https://github.com/hbldh/bleak/issues/972)).
> > That's why I switched to pygatt for now.
> 
> Hey Frans,
> 
> I've downloaded it and tested, but unforatuntly the state of the light entities stays off and is not switchable to on. It can find all the lamps and the addon installs fine, but they are not controlable.
> 
> https://i.gyazo.com/2dfc7eacb88a3faeb99f7f86934683bb.png
> 
> https://i.gyazo.com/d87cc96c109f40f5f0b117ecdd04b8f0.png

@0546timm can you check the logs and post them here? Check the readme for enabling debug logging https://github.com/fsaris/home-assistant-awox#troubleshooting

Did it work for you in a previous version?

**Comment by fsaris** [comment]
> Find devices process error: Command 'PATH=/usr/sbin:$PATH; rfkill unblock bluetooth' returned non-zero exit status 1."

This will be fixed in one of the next updates

**Comment by 0546timm** [comment]
> > > I created a beta release could you maybe test this (see https://hacs.xyz/docs/faq/beta)?
> > > I couldn't change to bleak (yet) as I'm running into an issue that the lights do not seem to follow the BLE standard ([hbldh/bleak#972](https://github.com/hbldh/bleak/issues/972)).
> > > That's why I switched to pygatt for now.
> > 
> > 
> > Hey Frans,
> > I've downloaded it and tested, but unforatuntly the state of the light entities stays off and is not switchable to on. It can find all the lamps and the addon installs fine, but they are not controlable.
> > https://i.gyazo.com/2dfc7eacb88a3faeb99f7f86934683bb.png
> > https://i.gyazo.com/d87cc96c109f40f5f0b117ecdd04b8f0.png
> 
> @0546timm can you check the logs and post them here? Check the readme for enabling debug logging https://github.com/fsaris/home-assistant-awox#troubleshooting
> 
> Did it work for you in a previous version?

Hey Frans,

Yes, been happily using your intergration for a year or so! Was a bit bazzled when it stopped working, so I'm very happy you are willing to put all this effort into this getting it to work, many kudos! 

Yep, lights worked fine, they did not change position or network, can control them with the Awox app. 
When enabling the debug and looking at the logs I can see it tries connecting but fails and then stops trying.

Below is for one of the lights, but they all do the same and throw the same error (same bulbs obviously).

```
2022-09-08 08:49:11.604 INFO (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:dc:0a:f5][Eettafel R] Failed to connect, trying next device []
2022-09-08 08:49:11.605 DEBUG (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:dc:0a:f5][Eettafel R] Setting up Bluetooth connection failed, making sure Bluetooth device stops trying
2022-09-08 08:49:11.613 DEBUG (SyncWorker_6) [custom_components.awox.awoxmeshlight] force stopping ble adapter
2022-09-08 08:49:11.917 INFO (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:2c:72:c9][Eettafel L] Trying to connect
2022-09-08 08:49:11.920 INFO (SyncWorker_1) [custom_components.awox.awoxmeshlight] reset bluetooth
2022-09-08 08:49:13.579 ERROR (SyncWorker_4) [custom_components.awox.awoxmeshlight] Timed out connecting to a4:c1:38:dc:0a:f5 after 10 seconds.
2022-09-08 08:49:13.776 INFO (SyncWorker_1) [custom_components.awox.awoxmeshlight] Connecting to a4:c1:38:2c:72:c9 with timeout=10
2022-09-08 08:49:21.919 ERROR (MainThread) [custom_components.awox.awox_mesh] exception 
Traceback (most recent call last):
  File "/config/custom_components/awox/awox_mesh.py", line 312, in _async_connect_device
    if await self.hass.async_add_executor_job(device.connect):
asyncio.exceptions.CancelledError
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/config/custom_components/awox/awox_mesh.py", line 311, in _async_connect_device
    async with async_timeout.timeout(10):
  File "/usr/local/lib/python3.10/site-packages/async_timeout/__init__.py", line 129, in __aexit__
    self._do_exit(exc_type)
  File "/usr/local/lib/python3.10/site-packages/async_timeout/__init__.py", line 212, in _do_exit
    raise asyncio.TimeoutError
asyncio.exceptions.TimeoutError
2022-09-08 08:49:21.926 INFO (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:2c:72:c9][Eettafel L] Failed to connect, trying next device []
2022-09-08 08:49:21.927 DEBUG (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:2c:72:c9][Eettafel L] Setting up Bluetooth connection failed, making sure Bluetooth device stops trying
2022-09-08 08:49:21.930 DEBUG (SyncWorker_5) [custom_components.awox.awoxmeshlight] force stopping ble adapter
2022-09-08 08:49:22.233 INFO (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:ef:41:9b][Plafondlamp slaapkamer] Trying to connect
2022-09-08 08:49:22.235 INFO (SyncWorker_7) [custom_components.awox.awoxmeshlight] reset bluetooth
2022-09-08 08:49:23.877 ERROR (SyncWorker_1) [custom_components.awox.awoxmeshlight] Timed out connecting to a4:c1:38:2c:72:c9 after 10 seconds.
2022-09-08 08:49:24.108 INFO (SyncWorker_7) [custom_components.awox.awoxmeshlight] Connecting to a4:c1:38:ef:41:9b with timeout=10

```

**Comment by fsaris** [comment]
@0546timm could you also enable logging for gattpy by adding `pygatt: debug` to the `log:` list in your config. And check the logs again

**Comment by 0546timm** [comment]
> t

Sure, here you go

`2022-09-08 12:15:03.483 INFO (MainThread) [custom_components.awox.awox_mesh] Search for AwoX devices to find closest (best RSSI value) device
2022-09-08 12:15:09.820 INFO (MainThread) [custom_components.awox.scanner] Scanning 10 seconds for AwoX bluetooth mesh devices!
2022-09-08 12:15:20.202 INFO (SyncWorker_1) [custom_components.awox.bluetoothctl] found: {'A4:C1:38:DC:0A:F5': {'mac': 'A4:C1:38:DC:0A:F5', 'name': 'A4:C1:38:DC:0A:F5', 'rssi': -74}, 'BC:7E:8B:F9:C8:EB': {'mac': 'BC:7E:8B:F9:C8:EB', 'name': 'BC:7E:8B:F9:C8:EB', 'rssi': -60}, 'E8:6F:38:0C:FF:84': {'mac': 'E8:6F:38:0C:FF:84', 'name': 'E8:6F:38:0C:FF:84', 'rssi': -83}, '7C:D1:C3:20:36:24': {'mac': '7C:D1:C3:20:36:24', 'name': '7C:D1:C3:20:36:24', 'rssi': -86}, '64:E7:D8:36:E4:BB': {'mac': '64:E7:D8:36:E4:BB', 'name': '64:E7:D8:36:E4:BB', 'rssi': -59}, 'A4:C1:38:2C:72:C9': {'mac': 'A4:C1:38:2C:72:C9', 'name': 'A4:C1:38:2C:72:C9', 'rssi': -79}}
2022-09-08 12:15:20.206 DEBUG (MainThread) [custom_components.awox.scanner] Found devices: {'A4:C1:38:DC:0A:F5': {'mac': 'A4:C1:38:DC:0A:F5', 'name': 'A4:C1:38:DC:0A:F5', 'rssi': -74}, 'A4:C1:38:2C:72:C9': {'mac': 'A4:C1:38:2C:72:C9', 'name': 'A4:C1:38:2C:72:C9', 'rssi': -79}}
2022-09-08 12:15:20.471 DEBUG (MainThread) [custom_components.awox.awox_mesh] Scan result: {'A4:C1:38:DC:0A:F5': {'mac': 'A4:C1:38:DC:0A:F5', 'name': 'A4:C1:38:DC:0A:F5', 'rssi': -74}, 'A4:C1:38:2C:72:C9': {'mac': 'A4:C1:38:2C:72:C9', 'name': 'A4:C1:38:2C:72:C9', 'rssi': -79}}
2022-09-08 12:15:20.472 INFO (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:dc:0a:f5][Eettafel R] Bluetooth scan returns RSSI value = -74
2022-09-08 12:15:20.472 INFO (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:2c:72:c9][Eettafel L] Bluetooth scan returns RSSI value = -79
2022-09-08 12:15:20.472 INFO (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:ef:41:9b][Plafondlamp slaapkamer] Device NOT found during Bluetooth scan
2022-09-08 12:15:20.473 INFO (MainThread) [custom_components.awox.awox_mesh] async_update: Request status
2022-09-08 12:15:20.473 INFO (MainThread) [custom_components.awox.awox_mesh] Queue command requestStatus {'dest': 65535, 'withResponse': True}
2022-09-08 12:15:20.476 DEBUG (AwoxMeshCommands-YVCnDHUh) [custom_components.awox.awox_mesh] process 0/0 - {'command': 'requestStatus', 'params': {'dest': 65535, 'withResponse': True}, 'callback': <function AwoxMesh._async_add_command_to_queue.<locals>.command_executed at 0x7f7fc99630>, 'allow_to_fail': False}
2022-09-08 12:15:20.495 INFO (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:dc:0a:f5][Eettafel R] Trying to connect
2022-09-08 12:15:20.496 INFO (SyncWorker_7) [custom_components.awox.awoxmeshlight] reset bluetooth
2022-09-08 12:15:20.497 DEBUG (SyncWorker_7) [pygatt.backends.gatttool.gatttool] gatttool_cmd=gatttool -i hci0 -I
2022-09-08 12:15:22.390 INFO (Thread-9) [pygatt.backends.gatttool.gatttool] Running...
2022-09-08 12:15:22.396 INFO (SyncWorker_7) [custom_components.awox.awoxmeshlight] Connecting to a4:c1:38:dc:0a:f5 with timeout=10
2022-09-08 12:15:30.496 ERROR (MainThread) [custom_components.awox.awox_mesh] exception 
Traceback (most recent call last):
  File "/config/custom_components/awox/awox_mesh.py", line 312, in _async_connect_device
    if await self.hass.async_add_executor_job(device.connect):
asyncio.exceptions.CancelledError
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/config/custom_components/awox/awox_mesh.py", line 311, in _async_connect_device
    async with async_timeout.timeout(10):
  File "/usr/local/lib/python3.10/site-packages/async_timeout/__init__.py", line 129, in __aexit__
    self._do_exit(exc_type)
  File "/usr/local/lib/python3.10/site-packages/async_timeout/__init__.py", line 212, in _do_exit
    raise asyncio.TimeoutError
asyncio.exceptions.TimeoutError
2022-09-08 12:15:30.509 INFO (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:dc:0a:f5][Eettafel R] Failed to connect, trying next device []
2022-09-08 12:15:30.509 DEBUG (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:dc:0a:f5][Eettafel R] Setting up Bluetooth connection failed, making sure Bluetooth device stops trying
2022-09-08 12:15:30.510 DEBUG (SyncWorker_8) [custom_components.awox.awoxmeshlight] force stopping ble adapter
2022-09-08 12:15:30.561 INFO (SyncWorker_8) [pygatt.backends.gatttool.gatttool] Stopping
2022-09-08 12:15:30.615 INFO (Thread-9) [pygatt.backends.gatttool.gatttool] Listener thread finished
2022-09-08 12:15:30.819 INFO (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:2c:72:c9][Eettafel L] Trying to connect
2022-09-08 12:15:30.822 INFO (SyncWorker_3) [custom_components.awox.awoxmeshlight] reset bluetooth
2022-09-08 12:15:30.823 DEBUG (SyncWorker_3) [pygatt.backends.gatttool.gatttool] gatttool_cmd=gatttool -i hci0 -I
2022-09-08 12:15:32.498 ERROR (SyncWorker_7) [custom_components.awox.awoxmeshlight] Timed out connecting to a4:c1:38:dc:0a:f5 after 10 seconds.
2022-09-08 12:15:32.716 INFO (Thread-10) [pygatt.backends.gatttool.gatttool] Running...
2022-09-08 12:15:32.717 INFO (SyncWorker_3) [custom_components.awox.awoxmeshlight] Connecting to a4:c1:38:2c:72:c9 with timeout=10
2022-09-08 12:15:40.474 ERROR (MainThread) [custom_components.awox.awox_mesh] Timeout fetching awox data
2022-09-08 12:15:40.477 DEBUG (MainThread) [custom_components.awox.awox_mesh] Finished fetching awox data in 36.994 seconds (success: False)
2022-09-08 12:15:40.822 ERROR (MainThread) [custom_components.awox.awox_mesh] exception 
Traceback (most recent call last):
  File "/config/custom_components/awox/awox_mesh.py", line 312, in _async_connect_device
    if await self.hass.async_add_executor_job(device.connect):
asyncio.exceptions.CancelledError
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/config/custom_components/awox/awox_mesh.py", line 311, in _async_connect_device
    async with async_timeout.timeout(10):
  File "/usr/local/lib/python3.10/site-packages/async_timeout/__init__.py", line 129, in __aexit__
    self._do_exit(exc_type)
  File "/usr/local/lib/python3.10/site-packages/async_timeout/__init__.py", line 212, in _do_exit
    raise asyncio.TimeoutError
asyncio.exceptions.TimeoutError
2022-09-08 12:15:40.833 INFO (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:2c:72:c9][Eettafel L] Failed to connect, trying next device []
2022-09-08 12:15:40.833 DEBUG (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:2c:72:c9][Eettafel L] Setting up Bluetooth connection failed, making sure Bluetooth device stops trying
2022-09-08 12:15:40.838 DEBUG (SyncWorker_2) [custom_components.awox.awoxmeshlight] force stopping ble adapter
2022-09-08 12:15:40.889 INFO (SyncWorker_2) [pygatt.backends.gatttool.gatttool] Stopping
2022-09-08 12:15:40.942 INFO (Thread-10) [pygatt.backends.gatttool.gatttool] Listener thread finished
2022-09-08 12:15:41.147 INFO (MainThread) [custom_components.awox.awox_mesh] [a4:c1:38:ef:41:9b][Plafondlamp slaapkamer] Trying to connect
2022-09-08 12:15:41.148 INFO (SyncWorker_0) [custom_components.awox.awoxmeshlight] reset bluetooth
2022-09-08 12:15:41.149 DEBUG (SyncWorker_0) [pygatt.backends.gatttool.gatttool] gatttool_cmd=gatttool -i hci0 -I
2022-09-08 12:15:42.818 ERROR (SyncWorker_3) [custom_components.awox.awoxmeshlight] Timed out connecting to a4:c1:38:2c:72:c9 after 10 seconds.
2022-09-08 12:15:43.029 INFO (Thread-11) [pygatt.backends.gatttool.gatttool] Running...
2022-09-08 12:15:43.030 INFO (SyncWorker_0) [custom_components.awox.awoxmeshlight] Connecting to a4:c1:38:ef:41:9b with timeout=10`

**Comment by fsaris** [comment]
@0546timm could you test the `main` version? Increased the timeout a little. What version of HA are you using?

**Comment by 0546timm** [comment]
> @0546timm could you test the `main` version? Increased the timeout a little. What version of HA are you using?

Working like a charm again! Many many thanks, you made my weekend already haha.

**Comment by OkOtChA666** [comment]
Sorry for the late reply.
It's working fine on the last version (September).
I ocasonaly lost connection on one lamp only. Everyhing else is fine. I can do what I did in the previously version.
I do get a lot of errors that appear on logs, but it still works fine.
Thank you.

**Comment by Dillton** [comment]
Hey Frans.

I was also on vacation so wasnt able to test the beta sooner.
So I am now running the latest HA + beta3 and everything seems to be working even with some errors in logs, I have also noticed that sometimes devices disconnects randomly for a few seconds. But I believe this will be fixed in future versions :).

Thank you very much and keep up the good work.

**Comment by netspherecyborg** [comment]
Thanks Dude! I am new to HA (yesterday old) but I managed to get also my bulbs (https://www.eglo.com/en/bulb-11659.html) to work with the beta. I have two of them and both of them work with RGB, dimmer and so on.

**Comment by Marsupilami78** [comment]
Heute wird eine neue Kernversion veröffentlicht, sobald ich diese installiere, machen die Lichter nur Probleme.

today a new core version is released, as soon as i install this the lights make only problems

**Comment by fsaris** [comment]
@Marsupilami78 did you use the latest beta version?

**Comment by Marsupilami78** [comment]
@fsaris 
i downloaded the beta version last night, unfortunately i can't see which one it was because i restored from backup. The problem occurred with the version, home assistenat core 2022.9.4

**Comment by Dillton** [comment]
Hi @Marsupilami78,
just updated to latest version and everything seems to be working.
![image](https://user-images.githubusercontent.com/68992112/190248004-f36b9de9-9b72-471c-88fd-52ceb48f83fc.png)

**Comment by Marsupilami78** [comment]
@Dillton 

is strange, with me there were so problems that the whole home assistant has started to lag.

**Comment by fsaris** [comment]
@Marsupilami78 you are sure this integration was the reason for lagging? Did you run it without this integration?

Could you enable debug logging I'm interested to know what part is causing the lag. And Maybe also check the processes on the host system to check if there is some bluetooth related process causing this.

It's important you use the latest beta to test because the first version didn't correctly disconnected from the GATTTOOL process.

**Comment by Marsupilami78** [comment]
@fsaris 

Ok, then I will try again. But it may take a few days, had a tooth surgery and severe pain

**Comment by Slion** [comment]
Also running 0.1.3b3 on HA 2022.9.4 and so far so good.
![image](https://user-images.githubusercontent.com/6508892/190856442-970bc684-c2fe-495b-a613-ab3fe61b6f5c.png)

I'm just getting started with it though so I can't tell how stable it will be. Tried one light and it worked.
Also tried [EGLO Connect Plug](https://www.amazon.de/-/en/gp/product/B07H1PZZT1/ref=ox_sc_act_title_1?smid=A1NI9PR7VS9H40&psc=1) and it worked too.

Planning to expose those through [Emulated Hue](https://www.home-assistant.io/integrations/emulated_hue/) for Logitech Harmony Hub to use them. Already did that with AVM Fritz!DECT 200 and that worked great so I reckon it should work too with Awox.

@fsaris Fantastic work you are doing here. Even more so since it looks like you had to re-implement it all over the summer.

**Comment by Slion** [comment]
> Any chance of having brightness control

Actually brightness control is already working from Logitech Harmony. Awesome stuff!

**Comment by Slion** [comment]
Ok, I have not tested all 7 lights I have yet but at least 3 are working but one is not, the toggle switch disabled for some reason. Can I force a new scan/discovery somehow?

![image](https://user-images.githubusercontent.com/6508892/190860799-c9473e8a-33f0-4f6d-9d34-e1a34e0b103a.png)

**Comment by Marsupilami78** [comment]
@fsaris 

now everything is running fine, thank you for your great work.

Home Assistant 2022.9.5
Supervisor 2022.09.1
Operating System 9.0
Frontend 20220907.2 - latest

**Comment by fsaris** [comment]
Resolved with release https://github.com/fsaris/home-assistant-awox/releases/tag/0.1.3

**Comment by 0546timm** [comment]
Hey @fsaris 

For some reason it can't find the BL Awox devices anymore. Even when I'm holding it next to the RPi.
Did you had any difficulties, maybe on the new HA version?

Logs just tell me it can't be found. I can see alot of devices it discovers but not my lightbulbs unfortunatly.

`2022-11-07 21:58:42.983 DEBUG (MainThread) [custom_components.awox.scanner] Found devices: {}
2022-11-07 21:58:43.272 DEBUG (MainThread) [custom_components.awox.awox_mesh] [YVCnDHUh] Scan result: {}
2022-11-07 21:58:43.273 INFO (MainThread) [custom_components.awox.awox_mesh] [YVCnDHUh][Terras][a4:c1:38:dc:0a:f5] Device NOT found during Bluetooth scan
2022-11-07 21:58:43.273 INFO (MainThread) [custom_components.awox.awox_mesh] [YVCnDHUh][Poort][a4:c1:38:2c:72:c9] Device NOT found during Bluetooth scan
2022-11-07 21:58:43.273 INFO (MainThread) [custom_components.awox.awox_mesh] [YVCnDHUh][Plafondlamp slaapkamer][a4:c1:38:ef:41:9b] Device NOT found during Bluetooth scan
2022-11-07 21:58:43.274 INFO (MainThread) [custom_components.awox.awox_mesh] [YVCnDHUh] Queue command requestStatus {'dest': 65535, 'withResponse': True}
2022-11-07 21:58:43.276 DEBUG (AwoxMeshCommands-YVCnDHUh) [custom_components.awox.awox_mesh] [YVCnDHUh] process 0/0 - {'command': 'requestStatus', 'params': {'dest': 65535, 'withResponse': True}, 'callback': <function AwoxMesh._async_add_command_to_queue.<locals>.command_executed at 0x7f9801a290>, 'allow_to_fail': False}
2022-11-07 21:58:43.285 WARNING (AwoxMeshCommands-YVCnDHUh) [custom_components.awox.awox_mesh] [YVCnDHUh] Command failed, retry 1
2022-11-07 21:58:43.296 WARNING (AwoxMeshCommands-YVCnDHUh) [custom_components.awox.awox_mesh] [YVCnDHUh] Command failed, retry 2
2022-11-07 21:58:43.302 WARNING (AwoxMeshCommands-YVCnDHUh) [custom_components.awox.awox_mesh] [YVCnDHUh] Command failed, retry 3
2022-11-07 21:58:43.307 DEBUG (AwoxMeshCommands-YVCnDHUh) [custom_components.awox.awox_mesh] [YVCnDHUh] get item from queue
2022-11-07 21:58:43.317 DEBUG (MainThread) [custom_components.awox.awox_mesh] Finished fetching awox data in 22.316 seconds (success: False)`

**Comment by 0546timm** [comment]
EDIT:
Got one ligt to connect but same as before it's dropping connection due to a timeout, is this something you can extend?

```
2022-11-07 22:10:41.676 INFO (MainThread) [custom_components.awox.awox_mesh] [YVCnDHUh][Plafondlamp slaapkamer][a4:c1:38:ef:41:9b] Trying to connect
2022-11-07 22:10:41.678 DEBUG (SyncWorker_5) [pygatt.backends.gatttool.gatttool] gatttool_cmd=gatttool -i hci0 -I
2022-11-07 22:10:43.377 INFO (Thread-9) [pygatt.backends.gatttool.gatttool] Running...
2022-11-07 22:10:43.383 INFO (SyncWorker_5) [custom_components.awox.awoxmeshlight] Connecting to a4:c1:38:ef:41:9b with timeout=15
2022-11-07 22:10:58.140 INFO (SyncWorker_5) [custom_components.awox.awoxmeshlight] [YVCnDHUh][a4:c1:38:ef:41:9b] Send pair message bytearray(b'\x0c\xe2Q\xe2\x01\xee\xc4y\xd2)S\xad-\xc6R\x14\xc8')
2022-11-07 22:10:58.140 DEBUG (SyncWorker_5) [pygatt.device] Looking up handle for characteristic 00010203-0405-0607-0809-0a0b0c0d1914
2022-11-07 22:10:58.999 INFO (Thread-9) [custom_components.awox.awoxmeshlight] [YVCnDHUh][a4:c1:38:ef:41:9b] Disconnected by backend
2022-11-07 22:11:01.667 WARNING (MainThread) [custom_components.awox.awox_mesh] [YVCnDHUh] Requesting status failed - [TimeoutError] 
2022-11-07 22:11:01.672 ERROR (MainThread) [custom_components.awox.awox_mesh] Error fetching awox data: No device connected
2022-11-07 22:11:01.677 DEBUG (MainThread) [custom_components.awox.awox_mesh] Finished fetching awox data in 44.098 seconds (success: False)
2022-11-07 22:11:01.684 INFO (MainThread) [custom_components.awox.awox_mesh] [YVCnDHUh][Plafondlamp slaapkamer][a4:c1:38:ef:41:9b] Failed to connect, trying next device [TimeoutError] 
2022-11-07 22:11:01.684 DEBUG (MainThread) [custom_components.awox.awox_mesh] [YVCnDHUh][Plafondlamp slaapkamer][a4:c1:38:ef:41:9b] Setting up Bluetooth connection failed, making sure Bluetooth device stops trying
2022-11-07 22:11:01.686 DEBUG (SyncWorker_7) [custom_components.awox.awoxmeshlight] [YVCnDHUh][a4:c1:38:ef:41:9b] Force stopping ble adapter
2022-11-07 22:11:01.686 INFO (SyncWorker_7) [pygatt.backends.gatttool.gatttool] Stopping
2022-11-07 22:11:01.740 INFO (Thread-9) [pygatt.backends.gatttool.gatttool] Listener thread finished
2022-11-07 22:11:01.939 WARNING (AwoxMeshCommands-YVCnDHUh) [custom_components.awox.awox_mesh] [YVCnDHUh] Command failed, retry 1
```
